src/services/meta.ts

```diff
@@ -1,28 +1,42 @@
 import { Service } from "../common";
-import { StrategiesMetadata, TokenMetadata, VaultMetadataOverrides } from "../types";
+import { Address, StrategiesMetadata, TokenMetadata, VaultMetadataOverrides } from "../types";
 
-const MetaURL = "https://meta.yearn.network";
-
-const CHAIN_ID_KEY = "{chain_id}";
+const META_URL = "https://meta.yearn.network";
 
 /**
  * [[MetaService]] fetches meta data about things such as vaults and tokens
  * from yearn-meta
  */
 export class MetaService extends Service {
-  async tokens(): Promise<TokenMetadata[]> {
-    return fetch(this.buildUrl(`tokens/${CHAIN_ID_KEY}/all`)).then(res => res.json());
+  async tokens(addresses?: Address[]): Promise<TokenMetadata[]> {
+    const tokensMetadata = await fetch(`${META_URL}/tokens/${this.chainId}/all`).then(res => res.json());
+
+    if (!addresses) {
+      return tokensMetadata;
+    }
+
+    return tokensMetadata.filter((tokenMetadata: TokenMetadata) => addresses.includes(tokenMetadata.address));
+  }
+
+  async token(address: Address): Promise<TokenMetadata> {
+    return fetch(`${META_URL}/tokens/${this.chainId}/${address}`).then(res => res.json());
```

Comment:
If meta does not exist for the token address, will this fail?

Reply:
Correct, but not gracefully. I think you're right, we should probably handle this a bit better.

What do you think of https://github.com/yearn/yearn-sdk/pull/263/commits/18a7032757a376dc00131f49dd09ae5773086a7c?